sw/device/tests/crypto/cryptotest/firmware/ecdh.c

```diff
@@ -54,14 +59,37 @@ status_t handle_ecdh(ujson_t *uj) {
       public_key.key = (uint32_t *)&pub_p256;
       key_config.key_length = P256_KEY_BYTES;
       shared_key_words = P256_KEY_BYTES / sizeof(uint32_t);
-      p256_masked_scalar_t private_key_masked;
-      memset(private_key_masked.share0, 0, kP256MaskedScalarShareBytes);
-      memcpy(private_key_masked.share0, uj_private_key.d0, kP256ScalarBytes);
-      memset(private_key_masked.share1, 0, kP256MaskedScalarShareBytes);
-      memcpy(private_key_masked.share1, uj_private_key.d1, kP256ScalarBytes);
-      private_key_masked_raw = (uint32_t *)&private_key_masked;
-      private_keyblob_length = sizeof(private_key_masked);
+      memset(private_key_masked_p256.share0, 0, kP256MaskedScalarShareBytes);
```

Comment:
Would it be possible to write a function that takes the same arguments as `memset` but that fills the buffer with random data instead of 0s?

Reply:
Yes, and in fact we already have one: https://github.com/lowRISC/opentitan/blob/7e2221507aeb1db27b2879ec74d2ad19061b37ad/sw/device/lib/base/hardened_memory.h#L60

It's slightly different than memset because it only works with aligned pointers, but it's good practice to handle secret values in words rather than bytes anyway.